### PR TITLE
minor cleanup of filestore initialization

### PR DESCRIPTION
**Pull request by endophage**
Signed-off-by: David Lawrence <david.lawrence@docker.com> (github: endophage)

### CODE REVIEW

**Comment by endophage** [comment]
@cyli @riyazdf completing the import/export refactor and merging KeyFileStore with KeyMemoryStore is going to be a lot more work than I want and will delay getting started on higher priority stuff. This is a minimal refactor that just cleans up a little duplicate code and starts laying out more clearly the interfaces. Lots more to be done but we can at least merge this.

Will probably have a minor testing drop just because most of the dupe code being deleted was covered so the ratio will get slightly worse.

**Comment by cyli** [comment]
Should we be removing some of the other `SimpleFileStore` code that's not being currently used?  e.g. `RemoveDir` and `ListDir` don't seem to be called anywhere except from tests.  Not sure if they're necessary for the later import/export refactor and merging.

Otherwise, LGTM pending CI!

File: trustmanager/filestore.go [hunk 1]
```diff
@@ -15,40 +15,33 @@ type SimpleFileStore struct {
 	perms   os.FileMode
 }
 
-// NewSimpleFileStore creates a directory with 755 permissions
-func NewSimpleFileStore(baseDir string, fileExt string) (*SimpleFileStore, error) {
+// NewFileStore creates a fully configurable file store
+func NewFileStore(baseDir, fileExt string, perms os.FileMode) (*SimpleFileStore, error) {
```
Comment (riyazdf) [inline trustmanager/filestore.go]: nit: should this method be unexported, and only be accessible from `NewSimpleFileStore` and `NewPrivateSimpleFileStore`?

**Comment by riyazdf** [comment]
LGTM pending nit comment and further potential cleanup on `SimpleFileStore` code that @cyli has suggested

File: trustmanager/filestore.go (hunk 1 above)
Comment (endophage) [inline trustmanager/filestore.go]: As the most general initializer I think it's semantically correct to export it. The other initializers are just helpful wrappers for our most common use cases.

Comment (riyazdf) [inline trustmanager/filestore.go]: Ok makes sense, this is good as is

**Comment by riyazdf** [comment]
not sure how to understand the codecov drop here, since we deleted so much unused code and only one new line is potentially uncovered.  Still LGTM, merging.

**Pull request merged**